src/dex/uniswap-v3/contract-math/Oracle.ts

```diff
@@ -23,7 +23,7 @@ export class Oracle {
   ): OracleObservation {
     const delta = blockTimestamp - last.blockTimestamp;
     return {
-      blockTimestamp: state.blockTimestamp,
+      blockTimestamp,
```

Comment:
Here we simply matched the .ts implementation with the .sol implementation: https://github.com/Uniswap/v3-core/blob/main/contracts/libraries/Oracle.sol#L39

You can see that new observations also update the blockTimestamp in the Solidity code.

Furthermore, since Uniswap's pricing does not depend on the oracle logic, you can be rest assured this won't negatively impact the Uniswap quoting mechanics.